tests/main/azure-cvm/task.yaml

```diff
@@ -0,0 +1,91 @@
+summary: Check installation of Azure CVM kernel
+
+details: |
+  This test checks that we are able to perform a single-boot installation of
+  Azure CVM kernel
+
+# Run this test for jammy+
+systems: [ubuntu-22.04, ubuntu-24*]
+
+environment:
+    SNAPD_DEB_FROM_REPO: false
+
+prepare: |
+  if [ "$SPREAD_REBOOT" -eq 1 ]; then
+    exit 0
+  fi
+
+  # Install the snapd build
+  dpkg -i "$GOHOME"/snapd_*.deb
+
+  # Build the initramfs package from the project if possible
+  # to detect any issues in the initramfs package that would affect
+  # the cloud kernel
+  if os.query is-ubuntu-ge 24.04; then
+      #shellcheck source=tests/lib/core-initrd.sh
+      . "$TESTSLIB"/core-initrd.sh
+      build_and_install_initramfs_deb
+  fi
+
+  # Otherwise install it
+  if os.query is-ubuntu-lt 24.04; then
+    # carries ubuntu-core-initramfs
+    quiet add-apt-repository ppa:snappy-dev/image -y
+    quiet apt update -y
+    quiet apt install ubuntu-core-initramfs -y
+  fi
+
+  # install the linux-firmware as the current version of
+  # ubuntu-core-initramfs does not depend on it, but nonetheless requires it
+  # to build the initrd, also install the linux-azure kernel
+  quiet apt install linux-firmware linux-azure -y
+
+  # reboot into the azure kernel
+  REBOOT
+
+execute: |
+  # verify after the first and second reboot that we are running an azure kernel
+  if [ "$SPREAD_REBOOT" -ge 1 ]; then
+    KERNEL_TYPE=$(uname -r | rev | cut -d'-' -f1 | rev)
+    if [ "$KERNEL_TYPE" != "azure" ]; then
+      echo "not running expected -azure kernel"
+      exit 1
+    fi
+
+    # on second reboot, we should still be running an azure kernel
+    if [ "$SPREAD_REBOOT" -eq 2 ]; then
+      exit 0
+    fi
+  fi
+
+  # build initramfs
+  ubuntu-core-initramfs create-initrd --kernelver "$(uname -r)"
+
+  # build UKI
+  ubuntu-core-initramfs create-efi --unsigned --kernelver "$(uname -r)" \
+    --cmdline "snapd_recovery_mode=cloudimg-rootfs console=tty1 console=ttyS0 earlyprintk=ttyS0 snapd.debug=1 systemd.journald.forward_to_console=1"
+
+  UKI_FILE="$(ls /boot/kernel.efi*-azure)"
+
+  # move UKI to ESP
+  mv "$UKI_FILE" /boot/efi/EFI/ubuntu
+
+  # ensure efivarfs is mounted
+  modprobe efivarfs || true
+  mount -t efivarfs efivarfs /sys/firmware/efi/efivars || true
+
+  # find either vda or sda
+  DISK_DEV=""
+  if [ -b /dev/vda ]; then
+    DISK_DEV="/dev/vda"
+  elif [ -b /dev/sda ]; then
+    DISK_DEV="/dev/sda"
+  else
+    echo "no suitable disk found for efibootmgr"
+    exit 1
+  fi
+
+  # add UKI to boot entries
+  KERNEL_EFI="$(basename "$UKI_FILE")"
+  efibootmgr --create --disk "$DISK_DEV" --part 15 --label "CVM UKI" --loader "/EFI/ubuntu/$KERNEL_EFI"
```

Comment:
Where is the 15 coming from? I see it in the script that was shared wit us, but not sure what it is about.

Reply:
I've updated this approach now, and removed the hardcoding.